Tools/CMake/AMReXTargetHelpers.cmake

```diff
@@ -107,3 +107,17 @@ function (set_cpp_sources_to_cuda_language _target)
    list(FILTER _sources INCLUDE REGEX "\\.cpp")
    set_source_files_properties(${_sources} PROPERTIES LANGUAGE CUDA )
 endfunction ()
+
+#
+# Setup an amrex-dependent target for cuda compilation.
+# This function ensures that the CUDA compilation of _target
+# is compatible with amrex CUDA build.
+#
+function (setup_target_for_cuda_compilation _target)
+   set_target_properties( ${_target}
+      PROPERTIES
+      CUDA_SEPARABLE_COMPILATION ON      # This adds -dc
+      CUDA_RESOLVE_DEVICE_SYMBOLS OFF
```

Comment:
I think this depends, if I am passing the WarpX executable target in here this will be problematic: https://cmake.org/cmake/help/v3.17/prop_tgt/CUDA_RESOLVE_DEVICE_SYMBOLS.html
```suggestion
```
But will test.

Reply:
We can toggle that property on and off depending on the nature of the target by checking:
```
get_target_property(_target_type ${_target} TYPE)
if (_target_type STREQUAL "EXECUTABLE")
  ...
endif ()
```

Reply:
I will also double check why we use it in mfix

Reply:
Yet... the actual default of this property that does exactly that.

Reply:
yeah...weird

Reply:
> I will also double check why we use it in mfix.

Just in case this helps: Looks like these defaults were not documented prior to CMake 3.14: https://cmake.org/cmake/help/v3.14/prop_tgt/CUDA_RESOLVE_DEVICE_SYMBOLS.html

Reply:
Tried now, definitely does not work on our executable and does not look safe to me to be changed from the default at this point.